lib_xua/src/core/mixer/mixer.xc

```diff
@@ -439,11 +442,52 @@ static void mixer1(chanend c_host, chanend c_mix_ctl, chanend ?c_mixer2, chanend
         /* Request from audio()/mixer2() */
         request = inuint(c_audio);
 
-        /* Forward on Request for data to decouple thread */
+        // Check if cmd is pending from last exchange with decoupler.
+        // If no command pending, transfer sample with audiohub to unblock it as
+        // soon as possible. This is esp important for the I2S TDM case where lots
+        // more samples need to be exchanged in lesser time than regular I2S
+        // (for example, for 96KHz tdm8. 4 i2s lines, exchange 32 samples in (1/(96000*8))seconds)
+        if(!cmd_pending_flag)
+        {
+            GiveSamplesToDevice(c_audio, samples_to_device_map);
+            GetSamplesFromDevice(c_audio);
+        }
+        else
+        {
+            // forward cmd from last exchange with decoupler to audio
+            switch(pending_cmd[0])
+            {
+                case XUA_AUDCTL_SET_SAMPLE_FREQ:
+                    outct(c_audio, pending_cmd[0]);
+                    outuint(c_audio, pending_cmd[1]);
+                    break;
+
+                case XUA_AUD_SET_AUDIO_START:
+                    /* Inform mixer2 (or audio()) about change */
+                    outct(c_audio, pending_cmd[0]);
+                    outuint(c_audio, pending_cmd[1]);
+                    outuint(c_audio, pending_cmd[2]);
+                    break;
+
+                case XUA_AUD_SET_AUDIO_STOP:
+                    /* Pass on command */
+                    outct(c_audio, pending_cmd[0]);
+                    break;
+
+                default:
+                    break;
+            }
+            chkct(c_audio, XS1_CT_END);
+            outct(c_host, XS1_CT_END); // Only now ack back to the decoupler.
+                                       // So in case of a command, the decoupler remains blocked for an extra sample period.
```

Comment:
Try to comment actual code, rather than a change - people reading this wont necessarily know what the old code did (or care)

Assuming "extra sample period" relates to the previous implementation..